leftwm-core/src/utils/helpers.rs

```diff
@@ -110,6 +110,10 @@ pub fn relative_find<T, F>(
 where
     F: Fn(&T) -> bool,
 {
+    if shift == 0 {
+        return list.get(0);
+    }
+
```

Comment:
I would avoid adding specific code to generic helpers. This may best be placed at the top of focus_window_change, where you focus the first non floating window in to_reorder (make sure if you exit early that to_reorder is appended back into state.windows, otherwise they will get lost and leftwm will crash).